module/meowfficer/collect.py

```diff
@@ -0,0 +1,190 @@
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            # End - Random left shift
+            if self.image_color_count(MEOWFFICER_SHIFT_DETECT,
+                                      color=MEOWFFICER_SHIFT_DETECT.color, threshold=221, count=650):
+                if not flag:
+                    confirm_timer.reset()
+                    flag = True
+                if confirm_timer.reached():
+                    break
+                continue
+
+            # End - No shift at all
+            if self.appear(MEOWFFICER_GET_CHECK, offset=(40, 40)):
+                if flag:
+                    confirm_timer.reset()
+                    flag = False
+                if confirm_timer.reached():
+                    break
+        return flag
+
+    def _meow_get_sr(self):
+        """
+        Handle SR cat acquisition
+        If has at least one unique skill
+        then lock to prevent used as feed
+
+        Pages:
+            in: MEOWFFICER_GET_CHECK
+            out: MEOWFFICER_GET_CHECK or MEOWFFICER_TRAIN
+        """
+        # Wait for complete load before examining skills
+        logger.info('SR cat detected, wait complete load and examine base skills')
+        grid = MEOWFFICER_SKILL_GRID_2 if self._meow_detect_shift() else MEOWFFICER_SKILL_GRID_1
+
+        # Appropriate grid acquired, scan for unique skills
+        has_unique = False
+        for _ in grid.buttons:
+            # Empty slot; check for many white pixels
+            if self.image_color_count(_, color=(255, 255, 247), threshold=221, count=200):
+                continue
+
+            # Non-empty slot; check for few white pixels
+            # i.e. roman numerals
+            if self.image_color_count(_, color=(255, 255, 255), threshold=221, count=25):
+                continue
+
+            # Detected unique skill; break
+            has_unique = True
+            break
+
+        # Execute appropriate route
+        # Transition into lock popup
+        logger.info('At least one unique skill detected; locking...') if has_unique else \
+            logger.info('No unique skills detected; skipping...')
+        self.ui_click(MEOWFFICER_TRAIN_CLICK_SAFE_AREA,
+                      appear_button=MEOWFFICER_GET_CHECK, check_button=MEOWFFICER_CONFIRM,
+                      offset=(40, 40), retry_wait=3, skip_first_screenshot=True)
+
+        # Transition out of lock popup
+        # Use callable as screen is variable
+        def check_popup_exit():
+            if self.appear(MEOWFFICER_GET_CHECK, offset=(40, 40)):
+                return True
+
+            if self.appear(MEOWFFICER_TRAIN_START, offset=(20, 20)):
+                return True
+
+            return False
+
+        self.ui_click(MEOWFFICER_CONFIRM if has_unique else MEOWFFICER_CANCEL,
+                      check_button=check_popup_exit, offset=(40, 20),
+                      retry_wait=3, skip_first_screenshot=True)
```

Comment:
I guess these are server specific, but would still like to confirm. CN has popups on SSR meowfficers only while SR don't. Do EN has popups on both SR and SSR meowfficers?